hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/BasicConnectionCreator.java

```diff
@@ -82,32 +80,24 @@ public Connection createConnection() {
 		return conn;
 	}
 
-	private ValueHolder<SQLExceptionConversionDelegate> simpleConverterAccess = new ValueHolder<SQLExceptionConversionDelegate>(
-			new ValueHolder.DeferredInitializer<SQLExceptionConversionDelegate>() {
-				@Override
-				public SQLExceptionConversionDelegate initialize() {
-					return new SQLExceptionConversionDelegate() {
-						private final SQLStateConversionDelegate sqlStateDelegate = new SQLStateConversionDelegate(
-								new ConversionContext() {
-									@Override
-									public ViolatedConstraintNameExtracter getViolatedConstraintNameExtracter() {
-										// this should never happen...
-										throw new HibernateException( "Unexpected call to org.hibernate.exception.spi.ConversionContext.getViolatedConstraintNameExtracter" );
-									}
-								}
-						);
-
-						@Override
-						public JDBCException convert(SQLException sqlException, String message, String sql) {
-							JDBCException exception = sqlStateDelegate.convert( sqlException, message, sql );
-							if ( exception == null ) {
-								// assume this is either a set-up problem or a problem connecting, which we will
-								// categorize the same here.
-								exception = new JDBCConnectionException( message, sqlException, sql );
-							}
-							return exception;
+	private ValueHolder<SQLExceptionConversionDelegate> simpleConverterAccess =
+			new ValueHolder<>( () -> new SQLExceptionConversionDelegate() {
+				private final SQLStateConversionDelegate sqlStateDelegate = new SQLStateConversionDelegate(
+						() -> {
+							// this should never happen...
+							throw new HibernateException( "Unexpected call to org.hibernate.exception.spi.ConversionContext.getViolatedConstraintNameExtracter" );
```

Comment:
maybe using Javadoc pattern is less confusing?
```
throw new HibernateException( "Unexpected call to org.hibernate.exception.spi.ConversionContext#getViolatedConstraintNameExtracter()" );
```

Reply:
Fixed